src/Symfony/Component/Validator/Constraints/TypeValidator.php

```diff
@@ -21,35 +21,38 @@
 class TypeValidator extends ConstraintValidator
 {
     private const VALIDATION_FUNCTIONS = [
-        'bool' => 'is_bool',
-        'boolean' => 'is_bool',
-        'int' => 'is_int',
-        'integer' => 'is_int',
-        'long' => 'is_int',
-        'float' => 'is_float',
-        'double' => 'is_float',
-        'real' => 'is_float',
-        'numeric' => 'is_numeric',
-        'string' => 'is_string',
-        'scalar' => 'is_scalar',
-        'array' => 'is_array',
-        'iterable' => 'is_iterable',
-        'countable' => 'is_countable',
-        'callable' => 'is_callable',
-        'object' => 'is_object',
-        'resource' => 'is_resource',
-        'null' => 'is_null',
-        'alnum' => 'ctype_alnum',
-        'alpha' => 'ctype_alpha',
-        'cntrl' => 'ctype_cntrl',
-        'digit' => 'ctype_digit',
-        'graph' => 'ctype_graph',
-        'lower' => 'ctype_lower',
-        'print' => 'ctype_print',
-        'punct' => 'ctype_punct',
-        'space' => 'ctype_space',
-        'upper' => 'ctype_upper',
-        'xdigit' => 'ctype_xdigit',
+      'bool' => 'is_bool',
+      'boolean' => 'is_bool',
+      'int' => 'is_int',
+      'integer' => 'is_int',
+      'long' => 'is_int',
+      'float' => 'is_float',
+      'double' => 'is_float',
+      'real' => 'is_float',
+      'number' => 'is_int || is_float',
```

Comment:
please mind the indentation
now I'm wondering: should number be defined as is_int || is_float && !is_nan?